### PR TITLE
Set environment to "production" by default.

### LINKED ISSUE #1003: Set environment by default: production
SDK guideline introduced: 

> Send an environment on each event. If none was detected or set by the user, production should be used.

See also:
* https://develop.sentry.dev/sdk/overview/#writing-an-sdk
* https://github.com/getsentry/sentry-dotnet/issues/549

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Set environment to "production" by default.

## :bulb: Motivation and Context

Fixes #1003

## :green_heart: How did you test it?

Unit tests.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Review by marandaneto — APPROVED** [review]
LGTM

**Review by bruno-garcia — APPROVED** [review]
Missing a changelog and we should also add this to the docs now though.
Both Android and Java will take this.

**Comment by maciejwalkowiak** [comment]
I'll add it to docs once 3.1.2 is released.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# 3.1.2
+
+* Enhancement: Set environment to "production" by default.
```
Comment (bruno-garcia) [inline CHANGELOG.md:3]:
```suggestion
* Enhancement: Set environment to "production" by default.

```

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**